src/shims/unix/linux/foreign_items.rs

```diff
@@ -79,13 +79,16 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
             "pthread_setname_np" => {
                 let [thread, name] =
                     this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
-                let res = this.pthread_setname_np(
+                let res = match this.pthread_setname_np(
                     this.read_scalar(thread)?,
                     this.read_scalar(name)?,
                     TASK_COMM_LEN,
                     /* truncate */ false,
-                )?;
-                let res = if res { Scalar::from_u32(0) } else { this.eval_libc("ERANGE") };
+                )? {
+                    ThreadNameResult::Ok => Scalar::from_u32(0),
+                    ThreadNameResult::NameTooLong => this.eval_libc("ERANGE"),
+                    ThreadNameResult::ThreadNotFound => this.eval_libc("ENOENT"),
```

Comment:
```suggestion
                    // Act like we faild to open `/proc/self/task/$tid/comm`.
                    ThreadNameResult::ThreadNotFound => this.eval_libc("ENOENT"),
```

Reply:
@YohDeadfall you marked this as resolved, but the comment isn't there. Please don't do that, it makes a lot of extra work to re-review the same thing again and again.